RequestService/RequestService.Repo/Repository.cs

```diff
@@ -718,19 +743,19 @@ public void ArchiveOldRequests(int daysSinceJobRequested, int daysSinceJobStatus
 
             foreach(Request r in requests)
             {
+                bool inactive = false;
                 foreach (EntityFramework.Entities.Job j in r.Job)
                 {
                     if ((JobStatuses)j.JobStatusId.Value == JobStatuses.Done || (JobStatuses)j.JobStatusId.Value == JobStatuses.Cancelled)
                     {
-                        bool inactive = j.RequestJobStatus.Min(x => (DateTime.Now.Date - x.DateCreated.Date).TotalDays > daysSinceJobStatusChanged);
-
-                        if (inactive && (r.Archive ?? false) == false)
-                        {
-                            r.Archive = true;
-                            _context.Request.Update(r);
-                        }
+                        inactive = j.RequestJobStatus.Min(x => (DateTime.Now.Date - x.DateCreated.Date).TotalDays > daysSinceJobStatusChanged);
                     }
                 }
+                if (inactive && (r.Archive ?? false) == false)
```

Comment:
At this point, I think inactive represents only the last job which is Done or Cancelled.  So, in a request with 2 jobs,  if job 1 is still active but job 2 is inactive then I think the request will (incorrectly) get archived.